inc/compatibility/woocommerce.php

```diff
@@ -155,6 +159,61 @@ public function init() {
 		$this->add_inline_selectors();
 	}
 
+	/**
+	 * Set a flag that tells the plugin that woocommerce pages were created from their tool.
+	 *
+	 * @return bool
+	 */
+	public function set_update_woo_width_flag() {
+
+		if ( ! isset( $_GET['page'] ) ) {
+			return false;
+		}
+
+		$current_page = sanitize_text_field( wp_unslash( $_GET['page'] ) );
+		if ( 'wc-status' !== $current_page && 'wc-setup' !== $current_page ) {
+			return false;
+		}
+
+		if ( $current_page === 'wc-status' && ( ! isset( $_GET['action'] ) || ! isset( $_GET['_wpnonce'] ) || 'install_pages' !== sanitize_text_field( wp_unslash( $_GET['action'] ) ) ) ) {
+			return false;
+		}
+
+		if ( $current_page === 'wc-setup' && ( ! isset( $_GET['step'] ) || 'payment' !== sanitize_text_field( wp_unslash( $_GET['step'] ) ) ) ) {
+			return false;
+		}
+
+		update_option( 'neve_update_woo_width', 'yes' );
+		set_transient( 'woocommerce_shop_page_id', 'executed', 10 * MINUTE_IN_SECONDS );
```

Comment:
do you know for what is this used?

Reply:
@selul when you go to WooCommerce -> Status -> Tools and you click on "Create default WooCommerce pages" they redirect you to a link like this `http://neve.local/wp-admin/admin.php?page=wc-status&tab=tools&action=install_pages&_wpnonce=fb96470544`. I could check the parameters and update pages smth like this:


```
if ( $current_page === 'wc-status' && ( ! isset( $_GET['action'] ) || ! isset( $_GET['_wpnonce'] ) || 'install_pages' !== sanitize_text_field( wp_unslash( $_GET['action'] ) ) ) ) {
	return false;
}
$shop_id     = get_option( 'woocommerce_shop_page_id' );
update_post_meta( $shop_id, 'neve_meta_sidebar', 'full-width' );
```

The problem is that after the redirect, $shop_id and other woo page ids are empty. So I set a flag to run the function a little bit later. I added the transient that it's there for 10 minutes, so if a user clicks again on "create pages" all those actions do not happen again.